.gitignore

```diff
@@ -3,3 +3,5 @@ binarybuilder
 solverdummy/*.log
 solverdummy/*.json
 solverdummy/precice-run/
+*.o
+*.so
```

Comment:
Do we really need to remove these? Does the build process no longer have these files?

Reply:
The testing process produces these files so they should be excluded from git